ReactWindows/ReactNative/UIManager/DependencyObjectExtensions.cs

```diff
@@ -97,6 +98,32 @@ internal static ThemedReactContext GetReactContext(this DependencyObject view)
             return elementData.Context;
         }
 
+        internal static void SetReactCompoundView(this DependencyObject view, IReactCompoundView compoundView)
+        {
+            if (view == null)
+                throw new ArgumentNullException(nameof(view));
+
+            s_properties.GetOrCreateValue(view).CompoundView = compoundView;
+        }
+
+        internal static IReactCompoundView GetReactCompoundView(this DependencyObject view)
+        {
+            if (view == null)
+                throw new ArgumentNullException(nameof(view));
+
+            var elementData = default(DependencyObjectData);
+            if (s_properties.TryGetValue(view, out elementData))
```

Comment:
We should add behavior to use a TypeAs expression to check if the DependencyObject itself is IReactCompoundView.